### PR TITLE
[TextField] consolidate se23 styles and logic

### LINKED ISSUE #9972: [TextField] Consolidate se23 logic and styles

### PR DESCRIPTION
**Pull request by gwyneplaine**
### WHY are these changes introduced?

Fixes #9972 

### WHAT is this pull request doing?

- `TextField` consolidate se23 logic
- `TextField` consolidate se23 styles

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [Storybook](https://5d559397bae39100201eedc1-ijxwodqxyd.chromatic.com/?path=/story/all-components-textfield--all)
- [Prod Storybook](https://storybook.polaris.shopify.com/?path=/story/all-components-textfield--all&globals=polarisSummerEditions2023:true)

### CODE REVIEW

File: polaris-react/src/components/TextField/TextField.scss [hunk 1]
```diff
@@ -17,11 +17,7 @@ $spinner-icon-size: 12px;
   cursor: text;
 
   svg {
-    fill: var(--p-color-icon);
-
-    #{$se23} & {
-      fill: var(--p-color-icon-subdued);
```
Comment (gwyneplaine) [inline polaris-react/src/components/TextField/TextField.scss:23]: This was previously incorrectly overriding the svg fill value on `disabled`.  Hence the storybook change. 
See [Figma](https://www.figma.com/file/jLLkmo9r28hiqPvf4s90E4/Polaris-Gen-3-Components-%5Balpha%5D?type=design&node-id=52147-22449&mode=design&t=qxZur6Fk6EpaoWVt-4)

See [Storybook changes](https://shopify.chromatic.com/build?appId=5d559397bae39100201eedc1&number=19629)

**Review by laurkim — APPROVED** [review]
Overall looks great 💯 

Small issues I noticed:
- Prefix has incorrect line height/font size for xs/sm breakpoints
- Borderless example in All storybook has incorrect line height for xs/sm breakpoints

**Comment by gwyneplaine** [comment]
> * Prefix has incorrect line height/font size for xs/sm breakpoints

Thanks for the catch on this one 🙏 , fixed 👍 

> * Borderless example in All storybook has incorrect line height for xs/sm breakpoints

Looks like the $se23 `.Input` styles in prod are overriding non uplift [`.borderless`](https://github.com/Shopify/polaris/blob/502411f373a7fa533c9028ecfd2f7d4958b7f641/polaris-react/src/components/TextField/TextField.scss#L219) styles. Not sure if this was intentional or a specificity regression during uplift. 

Prod: 
Screenshot 2023-08-22 at 9 44 13 am

This PR: 
Screenshot 2023-08-22 at 9 43 51 am

I'm gonna remove the line-height declaration in `.borderless` such that it inherits line-height changes in `.Input` since that seems to be in the spirit of the se23 changes, but happy to revert in a subsequent PR @laurkim  (cc @sarahill)

File: polaris-react/src/components/TextField/TextField.scss [hunk 2]
```diff
@@ -264,7 +216,6 @@ $spinner-icon-size: 12px;
   .Input,
   .Backdrop {
     border: none;
-    line-height: var(--p-font-line-height-1);
```
Comment (gwyneplaine) [inline polaris-react/src/components/TextField/TextField.scss:267]: Removed this line as it was already being overridden in pre-uplift due to specificity. 
This particular line height declaration also doesn't seem to make sense in the context of the new DL, taking into account the se23 changes that have been made.

**Pull request merged**